app/src/main/java/com/example/sharewhatyoucanproject/donor/DonorActivity.kt

```diff
@@ -0,0 +1,155 @@
+            intent.type = "image/*"
+            intent.action = Intent.ACTION_GET_CONTENT
+            startActivityForResult(Intent.createChooser(intent, "Select Picture"), 0)
+        }
+
+        arrayAdapter = ArrayAdapter<String>(
+            applicationContext,
+            android.R.layout.simple_spinner_dropdown_item,
+            foodTypes,
+        )
+        binding.spinner.adapter = arrayAdapter
+        binding.spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View,
+                position: Int,
+                id: Long,
+            ) {
+                foodTypes[position].also {
+                    donorViewModel.foodType = it
+                    applicationContext.showToast(it)
+                }
+            }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {}
+        }
+
+        binding.uploadbtn.setOnClickListener {
+            donorViewModel.title = binding.titleet.text.toString()
+            donorViewModel.description = binding.descet.text.toString()
+            donorViewModel.hourSet = binding.hourset.text.toString()
+
+            if (!donorViewModel.isDataCompleted()) {
+                applicationContext.showToast("Please Fill All Fields")
+                return@setOnClickListener
+            }
+
+            if (donorViewModel.isImageNotSelected()) {
+                applicationContext.showToast("Choose a image")
+                return@setOnClickListener
+            }
+
+            if (!hasPermissions(applicationContext, *locationPermissions)) {
+                applicationContext.showToast("Please Allow Location To Post")
+                ActivityCompat.requestPermissions(
+                    this@DonorActivity,
+                    locationPermissions,
+                    PERMISSION_ALL,
+                )
+                return@setOnClickListener
+            }
+
+            if (!isGPSEnabled(applicationContext)) {
+                applicationContext.showToast("Open Your GPS")
+                return@setOnClickListener
+            }
+
+            circularProgressIndicator.visibility = View.VISIBLE
+            if (donorViewModel.isFoodNotValid()) {
+                circularProgressIndicator.visibility = View.GONE
+                applicationContext.showToast("You Cannot Upload Food")
+                return@setOnClickListener
+            }
+
+            donorViewModel.getCurrentLocation()
+        }
+    }
+
+    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == 0) {
+            if (resultCode == RESULT_OK) {
+                if (data != null) {
+                    donorViewModel.filepath = data.data
+                    val bitmap = MediaStore.Images.Media.getBitmap(contentResolver, data.data)
```

Comment:
loading and displaying images can be very complex due to memory constraints on mobile devices. If this is a very large image it may crash due to an out of memory error exception. I recommend using one of the popular third party libraries for image loading and caching:

Picasso: https://square.github.io/picasso/ - this is older and well tested but does not have good interop with coroutines and compose
Coil: https://coil-kt.github.io/coil/ - this is newer and built on coroutines, which is more performant. Also has great interop for compose.

You do not need to change this as part of this PR, I would recommend integrating an image loading library as a separate PR.